src/openai/__init__.py

```diff
@@ -119,6 +124,8 @@
 
 api_key: str | None = None
 
+auth: AzureAuth | None = None
```

Comment:
I think we need an AuthLike protocol to avoid vanilla having azure sprinkled into its api surface area. The AzureAuth would be one such AuthLike thing.

Reply:
```python
class Auth(Protocol):
    def get_token(self) -> str: ...
```
?

Reply:
Or class `TokenAuth(Protocol): def get_token(self) -> str`.